packages/txservice/src/transaction.ts

```diff
@@ -0,0 +1,216 @@
+import { delay, jsonifyError } from "@connext/nxtp-utils";
+import { BigNumber, providers } from "ethers";
+import { BaseLogger } from "pino";
+
+import { TransactionServiceConfig } from "./config";
+import { ChainError } from "./error";
+import { ChainRpcProvider } from "./provider";
+import { FullTransaction, MinimalTransaction } from "./types";
+
+export class Transaction {
```

Comment:
Wow not even all the way through it, already love the separation

Reply:
Would be interesting to see it as an event emitter as well, especially since everything is event driven atm

Reply:
Can you give me either an example or an approximation of what events it should be emitting? Like should we just emit on send and confirm?

Reply:
Events to emit:
TRANSACTION_ATTEMPT_SUBMITTED (called on every gas bump)
TRANSACTION_CONFIRMED
TRANSACTION_FAILED

Check out [SDK](https://github.com/connext/nxtp/blob/main/packages/sdk/src/sdk.ts) for examples of how to set up the `attach` events. We like this much better than the standard node event emitter interface.

Reply:
Thank you @rhlsthrm , super helpful! Will get on this